network/topology/randomizedTopology.go

```diff
@@ -0,0 +1,165 @@
+package topology
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// RandomizedTopology generates a random topology per channel.
+// By random topology we mean a node is connected to any other co-channel nodes with some
+// edge probability.
+type RandomizedTopology struct {
+	me      flow.Identifier             // used to keep identifier of the node
+	state   protocol.State              // used to keep a read only protocol state
+	subMngr network.SubscriptionManager // used to keep track topics the node subscribed to
+	chance  uint64                      // used to translate connectedness probability into a number in [0, 100]
+	rng     random.Rand                 // used as a stateful random number generator to sample edges
+}
+
+// NewRandomizedTopology returns an instance of the RandomizedTopology.
+func NewRandomizedTopology(nodeID flow.Identifier, edgeProb float64, state protocol.State,
+	subMngr network.SubscriptionManager) (*RandomizedTopology, error) {
+	// edge probability should be a positive value between 0 and 1. However,
+	// we like it to be strictly greater than zero. Also, at the current scale of
+	// Flow, we need it to be greater than 0.01 to support probabilistic connectedness.
+	if edgeProb < 0.01 || edgeProb > 1 {
+		return nil, fmt.Errorf("randomized topology probability should in in range of [0.01, 1], wrong value: %f", edgeProb)
+	}
+
+	// generates seed and random number generator
+	seed, err := byteSeedFromID(nodeID)
```

Comment:
I guess using the node ID here doesn't have any particular property compared to using a random seed (generated locally), right?

Reply:
Right, however, at this point we just aim at it being deterministic. The next step would be to investigate using random beacon for sake of better entropy.

Reply:
Thanks for the answer. I think if we don't care about the determinism anymore, even using the output of the standard package `crypto/rand` would be fine.